src/main/java/com/comcast/zucchini/ZucchiniRuntime.java

```diff
@@ -197,14 +196,36 @@ public void run() throws IOException {
         TestContext tc = TestContext.getCurrent();
         AbstractZucchiniTest azt = tc.getParentTest();
 
+        Iterator<CucumberFeatureHolder> features = azt.fastrunIteratorFactory(this.ros.cucumberFeatures(this.rl).iterator());
+
         boolean parallel = azt.isParallel();
         boolean barrierEnabled = azt.canBarrier();
 
-        for(CucumberFeature cf : features) {
-            formatter.uri(cf.getPath());
-            formatter.feature(cf.getGherkinFeature());
+        CucumberFeatureHolder lastCFH = null;
+
+        while (true) {
```

Comment:
i'm a bit worried about `while(true)`. Wouldn't it be simpler if you just did: `while(features.hasNext())`? This way you could also drop the try/catch block